modules/router-store/src/router_selectors.ts

```diff
@@ -1,9 +1,21 @@
-import { createSelector } from '@ngrx/store';
+import {
+  createFeatureSelector,
+  createSelector,
+  MemoizedSelector,
+} from '@ngrx/store';
 import { RouterStateSelectors } from './models';
 import { RouterReducerState } from './reducer';
+import { DEFAULT_ROUTER_FEATURENAME } from './router_store_module';
+
+export function createRouterSelector<State extends {} = {}>(): MemoizedSelector<
```

Comment:
```suggestion
export function createRouterSelector<State extends Record<string, any>>(): MemoizedSelector<
```

or

```suggestion
export function createRouterSelector<State extends object>(): MemoizedSelector<
```

because empty object type `{}` is compatible with primitive values.

Reply:
Thanks, I'm going for the first one because `object` is giving lint warnings.